application.cpp

```diff
@@ -173,8 +193,48 @@ bool application::initialize_impl(int argc, char** argv, vector<abstract_plugin*
       write_default_config(config_file_name);
    }
 
-   bpo::store(bpo::parse_config_file<char>(config_file_name.make_preferred().string().c_str(),
-                                           my->_cfg_options, true), options);
+   bpo::parsed_options opts_from_config = bpo::parse_config_file<char>(config_file_name.make_preferred().string().c_str(), my->_cfg_options, true);
+   bpo::store(opts_from_config, options);
+
+   bool printed_unrecognized_header = false, printed_default_header = false;
+
+   std::vector<std::string> unknown_opts;
+   for(const bpo::basic_option<char>& opt : opts_from_config.options) {
+      if(opt.unregistered) {
+         if(!printed_unrecognized_header)
+            std::cerr << "WARNING: The following options have been specified in the config file, but are unrecognized. Ignoring them." << std::endl;
+         printed_unrecognized_header = true;
+         std::cerr << "   " << opt.string_key << std::endl;
+      }
+   }
+
+   for(const boost::shared_ptr<bpo::option_description>& od_ptr : my->_cfg_options.options()) {
+      boost::any default_val, config_val;
+      if(!od_ptr->semantic()->apply_default(default_val))
+         continue;
+
+      for(const bpo::basic_option<char>& opt : opts_from_config.options) {
+         if(opt.string_key != od_ptr->long_name())
+            continue;
+
+         od_ptr->semantic()->parse(config_val, opt.value, true);
+         try {
+            if(my->_any_compare_map.at(default_val.type())(default_val, config_val)) {
+               if(!printed_default_header)
+                  std::cerr << "Warning: The following configuration items in the config.ini file are redundantly set to their default value." << std::endl
+                            << "Ensure this is what you absolutely want as future changes to application defaults will be overruled in favor" << std::endl
```

Comment:
Not sure this really conveys what you mean.  How about: "Explicit values override future changes to application defaults.  Consider commenting out or removing these items."

Reply:
I changed the text to incorporate your feedback. Also, by wrapping the config items across multiple lines it prevents the spammyness a little (the example I've put the in the PR is now exhaustive of what it would look like for a user)